zenoh-jni/src/key_expr.rs

```diff
@@ -152,3 +115,74 @@ pub(crate) unsafe extern "C" fn Java_io_zenoh_jni_JNIKeyExpr_freePtrViaJNI(
 ) {
     Arc::from_raw(ptr);
 }
+
+pub(crate) fn decode_key_expr(env: &mut JNIEnv, key_expr: &JString) -> Result<KeyExpr<'static>> {
+    let key_expr_str = decode_string(env, key_expr).map_err(|err| {
+        Error::Jni(format!(
+            "Unable to get key expression string value: {}",
+            err
+        ))
+    })?;
+    let key_expr = KeyExpr::try_from(key_expr_str).map_err(|err| {
+        Error::Jni(format!(
+            "Unable to create key expression from string: {}",
+            err
+        ))
+    })?;
+    Ok(key_expr)
+}
+
+pub(crate) fn autocanonize_key_expr(
+    env: &mut JNIEnv,
+    key_expr: &JString,
+) -> Result<KeyExpr<'static>> {
+    let key_expr_str = decode_string(env, key_expr).map_err(|err| {
+        Error::Jni(format!(
+            "Unable to get key expression string value: {}",
+            err
+        ))
+    })?;
+    let key_expr = KeyExpr::autocanonize(key_expr_str).map_err(|err| {
+        Error::Jni(format!(
+            "Unable to create key expression from string: {}",
+            err
+        ))
+    })?;
+    Ok(key_expr)
+}
+
+pub(crate) unsafe fn process_key_expr(
+    env: &mut JNIEnv,
+    key_expr_str: &JString,
+    key_expr_ptr: *const KeyExpr<'static>,
+) -> Result<KeyExpr<'static>> {
+    if key_expr_ptr.is_null() {
+        match decode_key_expr(env, key_expr_str) {
+            Ok(key_expr) => Ok(key_expr),
+            Err(err) => Err(Error::Jni(format!(
+                "Unable to process key expression: {}",
+                err
+            ))),
+        }
```

Comment:
improvement: 
when Mapping `Ok(x) => Ok(x)`, a nice way to reduce some lines of code is to use the `map_err` 
function as we only really want to change what the error outputs.
Example below
```rust 
decode_key_expr(env, key_expr_str)
    .map_err(|err| Error::Jni(format!("Unable to process key expression: {}", err)))
```

Reply:
Right.